docs/AQL/grammar/AqlParser.g4

```diff
@@ -0,0 +1,214 @@
+//
+//  description:  ANTLR4 parser grammar for Archetype Query Language (AQL)
+//  author:       Sebastian Iancu
+//  contributors: This version of the grammar is a complete rewrite of previously published antlr3 grammar,
+//                based on current AQL specifications in combination with other grammars from several AQL implementations.
+//                The openEHR Foundation would like to recognise the following people for their contributions:
+//                  - Chunlan Ma & Heath Frankel, Ocen Health Systems, Australia
+//                  - Bostjan Lah, Better, Slovenia
+//                  - Christian Chevalley, EHRBase, Germany
+//                  - Teun van Hemert & Michael Böckers, Nedap, Netherlands
+//  support:      openEHR Specifications PR tracker <https://specifications.openehr.org/releases/QUERY/open_issues>
+//  copyright:    Copyright (c) 2020 openEHR Foundation
+//  license:      Apache 2.0 License <http://www.apache.org/licenses/LICENSE-2.0.html>
+//
+
+parser grammar AqlParser;
+
+options { tokenVocab=AqlLexer; }
+
+selectQuery
+    : selectClause fromClause whereClause? orderByClause? limitClause? MINUSMINUS? EOF
+    ;
+
+selectClause
+    : SELECT top? selectExpr (COMMA selectExpr)*
+    ;
+
+fromClause
+    : FROM fromExpr
+    ;
+
+whereClause
+    : WHERE whereExpr
+    ;
+
+orderByClause
+    : ORDER BY orderByExpr (COMMA orderByExpr)*
+    ;
+
+limitClause
+    : LIMIT limit=NN_INTEGER (OFFSET offset=NN_INTEGER)?
+    ;
+
+
+selectExpr
+    : columnExpr (AS IDENTIFIER)?
+    ;
+
+fromExpr
+    : containsExpr
+    ;
+
+whereExpr
+    : NOT? identifiedExpr
+    | whereExpr AND whereExpr
+    | whereExpr OR whereExpr
+    | OPEN_PAR whereExpr CLOSE_PAR
+    ;
+
+orderByExpr
+    : identifiedPath order=(DESCENDING|DESC|ASCENDING|ASC)?
+    ;
+
+
+columnExpr
+    : identifiedPath
+    | aggregateFunctionCall
+    | functionCall
+    ;
+
+containsExpr
+    : classExprOperand (CONTAINS containsExpr)?
+    | containsExpr AND containsExpr
+    | containsExpr OR containsExpr
+    | OPEN_PAR containsExpr CLOSE_PAR
+    ;
+
+identifiedExpr
+    : EXISTS identifiedPath
+    | identifiedPath COMPARISON_OPERATOR identifiedOperand
+    | functionCall COMPARISON_OPERATOR identifiedOperand
```

Comment:
@teunvanhemert, should/can this be written in a single rule?

My intention here was to capture COMPARISON_OPERATOR only when left operand is either an identifiedPath or a function call, but not primitives, parameters, functions, etc.

Reply:
No this is fine like this.